extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -32,21 +36,182 @@ const fontsToLoad = [
+
+    // TODO: Remove support for url queryParam when widely available.
+    /** @private {string} */
+    this.storyUrl_ = hashParams['url'] || this.win.location.href.split('#')[0];
+
+    /** @private {!DevToolsTab} get URL param for tab (eg: #tab=page-experience) or default to PREVIEW*/
+    this.currentTab_ =
+      Object.values(DevToolsTab).find(
+        (e) => e.toLowerCase().replace(' ', '-') === hashParams['tab']
+      ) || DevToolsTab.PREVIEW;
+
+    /** @private {!Object} maps tabs to contents */
+    this.tabContents_ = Object.values(DevToolsTab).reduce(
+      (tabContents, tab) => {
+        tabContents[tab] = createTabElement(this.win, this.storyUrl_, tab);
+        return tabContents;
+      },
+      {}
+    );
   }
 
   /** @override */
   buildCallback() {
     this.loadFonts_();
+    this.buildLayout_();
+    this.initializeListeners_();
+
+    this.switchTab_(this.currentTab_);
+  }
+
+  /**
+   * @private
+   */
+  initializeListeners_() {
+    const tabsContainer = this.element.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    tabsContainer.addEventListener('click', (event) => {
+      const tab = event.target.textContent;
+      if (
+        Object.values(DevToolsTab).find((t) => t === tab) &&
+        this.currentTab_ != tab
+      ) {
+        this.switchTab_(tab);
+        // Update hashString with tab selected or null if tab = preview (default)
+        updateHash(
+          {
+            'tab':
+              tab == DevToolsTab.PREVIEW
+                ? null
+                : tab.toLowerCase().replace(' ', '-'),
+          },
+          this.win
+        );
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  buildLayout_() {
+    const container = buildContainerTemplate(this.element);
+    this.element.appendChild(container);
+    this.element.querySelector(
+      '.i-amphtml-story-dev-tools-close'
+    ).href = this.storyUrl_;
 
-    // TODO(mszylkowski): Create actual layout.
-    const message = this.element.ownerDocument.createElement('div');
-    message.textContent = 'AMP Story Dev-tools was loaded, page WIP';
-    this.element.textContent = '';
-    this.element.appendChild(message);
+    // Create tabs on top
+    const tabsContainer = container.querySelector(
+      '.i-amphtml-story-dev-tools-tabs'
+    );
+    Object.values(DevToolsTab).forEach((tabTitle) => {
+      const tab = this.win.document.createElement('button');
+      tab.textContent = tabTitle;
+      tabsContainer.appendChild(tab);
+    });
+  }
+
+  /**
+   * Switches the tab shown by activating the tab button and switching the contents.
+   * @param {!DevToolsTab} tab
+   */
+  switchTab_(tab) {
+    this.tabContents_[this.currentTab_].remove();
+    this.element
+      .querySelector('.i-amphtml-story-dev-tools-container')
+      .appendChild(this.tabContents_[tab]);
+    toArray(
+      this.element.querySelector('.i-amphtml-story-dev-tools-tabs').children
```

Comment:
You're making assumptions on what the DOM looks like which is very fragile, please give a class to your tabs and query them directly.